x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.js

```diff
@@ -811,45 +811,71 @@ export class DataRecognizer {
   }
 
   applyJobConfigOverrides(moduleConfig, jobOverrides, jobPrefix = '') {
-    if(jobOverrides !== undefined && jobOverrides !== null) {
-      if (typeof jobOverrides !== 'object') {
-        throw Boom.badRequest(
-          `Incompatible jobOverrides type (${typeof jobOverrides}). It needs to be an object or array of objects.`
-        );
+    if (jobOverrides === undefined || jobOverrides === null) {
+      return;
+    }
+
+    if (typeof jobOverrides !== 'object') {
+      throw Boom.badRequest(
+        `Incompatible jobOverrides type (${typeof jobOverrides}). It needs to be an object or array of objects.`
+      );
+    }
+
+    // jobOverrides could be a single object or an array of objects.
+    // if single, convert to an array
+    const overrides = Array.isArray(jobOverrides) ? jobOverrides : [jobOverrides];
+    const { jobs } = moduleConfig;
+
+    // separate all the overrides.
+    // the overrides which don't contain a job id will be applied to all jobs in the module
+    const generalOverrides = [];
+    const jobSpecificOverrides = [];
+
+    overrides.forEach(override => {
+      if (override.job_id === undefined) {
+        generalOverrides.push(override);
+      } else {
+        jobSpecificOverrides.push(override);
       }
+    });
 
-      // jobOverrides could be a single object or an array of objects.
-      // if single, convert to an array
-      const overrides = Array.isArray(jobOverrides) ? jobOverrides : [jobOverrides];
-      const { jobs } = moduleConfig;
+    function processArrayValues(source, update) {
+      if (typeof source !== 'object' || typeof update !== 'object') {
+        return;
+      }
+      Object.keys(source).forEach(key => {
+        const value = source[key];
 
-      // separate all the overrides.
-      // the overrides which don't contain a job id will be applied to all jobs in the module
-      const generalOverrides = [];
-      const jobSpecificOverrides = [];
-      overrides.forEach(o => {
-        if (o.job_id === undefined) {
-          generalOverrides.push(o);
+        if (typeof value !== 'object' || value === null) {
+          return;
+        }
+
+        if (!Array.isArray(value)) {
+          processArrayValues(value, update[key]);
         } else {
-          jobSpecificOverrides.push(o);
+          source[key] = update[key];
```

Comment:
this will wipe the source property if there is no corresponding property in the update.
`update[key]` could be `undefined` because the user isn't trying to override it.

Reply:
please check https://github.com/elastic/kibana/pull/49310/commits/661d8afe8e7c5737d6d08100606eea420f17006c